sdks/java/io/bigquery-io-perf-tests/src/test/java/org/apache/beam/sdk/bigqueryioperftests/BigQueryIOReadPerformanceIT.java

```diff
@@ -0,0 +1,200 @@
+    bigQueryMetricsDataset = options.getBigQueryDataset();
+    bigQueryMetricsTable = options.getBigQueryTable();
+    bigQuerySourceDataset = options.getBigQuerySourceDataset();
+    bigQuerySourceTable = options.getBigQuerySourceTable();
+    BigQueryOptions bigQueryOptions = BigQueryOptions.newBuilder().build();
+    tableQualifier =
+        String.format(
+            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    if (!checkForSourceTable()) {
+      createAndFillSourceTable();
+    }
+  }
+
+  private static void createAndFillSourceTable() {
+
+    Pipeline pipeline = Pipeline.create();
+
+    pipeline
+        .apply("Read from source", Read.from(new SyntheticBoundedSource(sourceOptions)))
+        .apply("Map records", ParDo.of(new MapKVToV()))
+        .apply(
+            "Write to BQ",
+            BigQueryIO.<byte[]>write()
+                .to(tableQualifier)
+                .withFormatFunction(
+                    input -> {
+                      TableRow tableRow = new TableRow();
+                      tableRow.set("data", input);
+                      return tableRow;
+                    })
+                .withCustomGcsTempLocation(ValueProvider.StaticValueProvider.of(tempRoot))
+                .withSchema(
+                    new TableSchema()
+                        .setFields(
+                            Collections.singletonList(
+                                new TableFieldSchema().setName("data").setType("BYTES")))));
+    pipeline.run().waitUntilFinish();
+  }
+
+  private static boolean checkForSourceTable() {
+    BigQueryOptions options = BigQueryOptions.newBuilder().build();
+    BigQuery client = options.getService();
+    TableId tableId =
+        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    Table sourceTable = client.getTable(tableId, TableOption.fields(BigQuery.TableField.TYPE));
+    return sourceTable != null;
+  }
+
+  @Test
+  public void testRead() {
+    Pipeline pipeline = Pipeline.create(options);
+    pipeline
+        .apply("Read from BQ", BigQueryIO.readTableRows().from(tableQualifier))
+        .apply("Gather time", ParDo.of(new TimeMonitor<>(NAMESPACE, "read_time")));
+    PipelineResult result = pipeline.run();
+    result.waitUntilFinish();
+    gatherAndPublishMetrics(result);
+  }
+
+  private void gatherAndPublishMetrics(PipelineResult result) {
+    String uuid = UUID.randomUUID().toString();
+    String timestamp = Timestamp.now().toString();
+    Set<Function<MetricsReader, NamedTestResult>> metricSuppliers =
+        getMetricSuppliers(uuid, timestamp);
+    IOITMetrics metrics = new IOITMetrics(metricSuppliers, result, NAMESPACE, uuid, timestamp);
+    metrics.publish(bigQueryMetricsDataset, bigQueryMetricsTable);
+  }
+
+  private Set<Function<MetricsReader, NamedTestResult>> getMetricSuppliers(
```

Comment:
There's only one supplier. Can we just call supplier.apply() without creating the list of them? I did something similar in [KafkaIOIT](https://github.com/apache/beam/blob/master/sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOIT.java#L138)

Reply:
And then we can publish outside the method so that the method does not do 2 things (gather(), publish() rather thant gatherAndPublish() in one method).

Reply:
That's a good idea, thanks!